tests/Transforms/validate_noise/validate_noise_preserve_user_param_fail.mlir

```diff
@@ -0,0 +1,9 @@
+// RUN: heir-opt --mlir-to-secret-arithmetic --secret-insert-mgmt-bgv --validate-noise=model=bgv-noise-by-bound-coeff-average-case-pk %s --verify-diagnostics
+
+// expected-error@below {{'builtin.module' op Noise validation failed.}}
+module attributes {bgv.schemeParam = #bgv.scheme_param<logN = 11, Q = [17], P = [1093633], plaintextModulus = 65537>} {
```

Comment:
```suggestion
module attributes {bgv.schemeParam = #bgv.scheme_param<logN = 11, Q = [17, 18], P = [1093633], plaintextModulus = 65537>} {
```

I think the problem is here: the default scheme parameters generated in this case has only one logqi, but the [`getCurrentLevel` indexed here](https://github.com/google/heir/blob/fa56d4116a188911970b4a7fce8c7ebe7669d9b1/lib/Analysis/NoiseAnalysis/BGV/NoiseByBoundCoeffModel.cpp#L187) is 1. Adding a second entry to Q seems to fix the problem, but maybe you want to patch the code to be more robust to this edge case.

Reply:
OK I should then verify the user provided param has the same `Level` as the max level of the `MgmtAttr`. Should I push another version or open another PR?

Reply:
Also maybe we can add ASAN to the dev config/CI?

Reply:
> OK I should then verify the user provided param has the same `Level` as the max level of the `MgmtAttr`. Should I push another version or open another PR?

You can amend the commit in this PR with your fix, it will propagate internally.

Reply:
pushed the fixed version with both fail noise validation test and fail level requirement test. In the future we should add more test/sanitizer on user input parameter.